File: blog/20150507/spring_switch_controller/src/main/java/sample/controller/SampleController.java
package sample.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
public class SampleController {
	@RequestMapping("/sample/{id}")
	@ResponseBody
	public String sample(@PathVariable("id") String id) {
		return "sample: " + id + ", " + System.currentTimeMillis();
	}
}

File: kotlin/spring/webflux/session_sample/src/main/kotlin/sample/controller/SampleController.kt
package sample.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.server.WebSession
import reactor.core.publisher.Mono

@RestController
class SampleController {
    @GetMapping("/set/{value}")
    fun set(@PathVariable("value") value: String, session: WebSession): Mono<Data> {
        session.attributes["value"] = value
        return Mono.just(Data(value))
    }

    @GetMapping("/get")
    fun get(session: WebSession): Mono<Data> {
        val value = session.attributes["value"].toString()
        return session.invalidate().thenReturn(Data(value))
    }
}

data class Data(val value: String)